ipld/bitfield/src/lib.rs

```diff
@@ -351,14 +358,76 @@ impl BitOr<&BitField> for &BitField {
 
     #[inline]
     fn bitor(self, rhs: &BitField) -> Self::Output {
-        BitField::from_ranges(self.ranges().union(rhs.ranges()))
+        if self.is_trivially_empty() {
+            rhs.clone()
+        } else if rhs.is_trivially_empty() {
+            self.clone()
+        } else {
+            BitField::from_ranges(self.ranges().union(rhs.ranges()))
+        }
+    }
+}
+
+impl BitOr<BitField> for &BitField {
+    type Output = BitField;
+
+    #[inline]
+    fn bitor(self, rhs: BitField) -> Self::Output {
+        if self.is_trivially_empty() {
+            rhs
+        } else if rhs.is_trivially_empty() {
+            self.clone()
+        } else {
+            BitField::from_ranges(self.ranges().union(rhs.ranges()))
+        }
+    }
+}
+
+impl BitOr<&BitField> for BitField {
+    type Output = BitField;
+
+    #[inline]
+    fn bitor(self, rhs: &BitField) -> Self::Output {
+        if rhs.is_trivially_empty() {
```

Comment:
nit: unless I'm missing some speedup gain it would be nice to check in a consistent order.  Other bitors do self, then rhs

Reply:
I'm trying to avoid a clone if both are trivially empty. It probably won't matter in most case so I'll switch it back.